specs/builder.go

```diff
@@ -0,0 +1,256 @@
+				Length:      dummyField.Length,
+				IDLength:    dummyField.IDLength,
+				Description: dummyField.Description,
+				Enc:         enc,
+				Pref:        pref,
+				Pad:         pad,
+			})
+		case reflect.TypeOf(field.Binary{}).Name():
+			newField = field.NewBinary(&field.Spec{
+				Length:      dummyField.Length,
+				IDLength:    dummyField.IDLength,
+				Description: dummyField.Description,
+				Enc:         enc,
+				Pref:        pref,
+				Pad:         pad,
+			})
+		case reflect.TypeOf(field.Bitmap{}).Name():
+			newField = field.NewBitmap(&field.Spec{
+				Length:      dummyField.Length,
+				IDLength:    dummyField.IDLength,
+				Description: dummyField.Description,
+				Enc:         enc,
+				Pref:        pref,
+				Pad:         pad,
+			})
+		default:
+			continue
+		}
+
+		spec.Fields[index] = newField
+	}
+
+	return &spec, nil
+}
+
+func (builder *messageSpecBuilder) ExportJSON(orgSpec *iso8583.MessageSpec) ([]byte, error) {
+
+	if orgSpec == nil {
+		return nil, fmt.Errorf("invalid message spec")
+	}
+
+	dummyJson := specDummy{
+		Name:   orgSpec.Name,
+		Fields: make(map[string]fieldDummy),
+	}
+
+	for index, orgField := range orgSpec.Fields {
+		dummyName := fmt.Sprintf(FieldPrefix+"%03d", index)
+		dummyField := fieldDummy{
+			Type: reflect.TypeOf(orgField).Elem().Name(),
+		}
+
+		switch reflect.TypeOf(orgField).Elem() {
+		case reflect.TypeOf(field.String{}), reflect.TypeOf(field.Numeric{}), reflect.TypeOf(field.Binary{}), reflect.TypeOf(field.Bitmap{}):
+			spec := orgField.Spec()
+
+			dummyField.Length = spec.Length
+			dummyField.IDLength = spec.IDLength
+			dummyField.Description = spec.Description
+			if spec.Enc != nil {
+				dummyField.Enc = reflect.TypeOf(spec.Enc).Elem().Name()
+			}
+			if spec.Pref != nil {
+				dummyField.Prefix = reflect.TypeOf(spec.Pref).Elem().Name()
+
+				var lengthStr, str1, str2 string
+				if fmt.Sscanf(spec.Pref.Inspect(), "%s %s %s", &str1, &lengthStr, &str2); len(lengthStr) > 0 && lengthStr != "fixed" && strings.ToUpper(lengthStr) == strings.Repeat("L", len(lengthStr)) {
+					dummyField.PrefixLength = len(lengthStr)
+				}
+			}
+			if spec.Pad != nil {
+				pad := padDummy{}
+				switch reflect.TypeOf(spec.Pad).Elem().String() {
+				case "padding.leftPadder":
+					if spec.Pad.Inspect() != nil {
+						pad.Type = reflect.TypeOf(spec.Pad).Elem().String()
+						pad.Pad, _ = utf8.DecodeRune(spec.Pad.Inspect())
+					}
+				}
+				if pad.Type != "" {
+					dummyField.Padding = &pad
+				}
+			}
+
+		default:
+			continue
+		}
+
+		dummyJson.Fields[dummyName] = dummyField
+	}
+
+	outputBuffer := new(bytes.Buffer)
+	enc := json.NewEncoder(outputBuffer)
+	enc.SetEscapeHTML(false)
+	enc.SetIndent("", "\t")
+	err := enc.Encode(dummyJson)
+
+	if err != nil {
+		return nil, fmt.Errorf("unable to export message spec")
+	}
+
+	return outputBuffer.Bytes(), nil
+}
+
+func getPrefixInterface(prefixName string, length int) prefix.Prefixer {
+
+	var prefixes = map[string]prefix.Prefixer{
```

Comment:
to not re-define it on every call of the function, we can move the var definition outside of the func.